src/UriFactory.php

```diff
@@ -16,4 +36,217 @@ public function createUri(string $uri = ''): UriInterface
     {
         return new Uri($uri);
     }
+
+    /**
+     * Create a Uri instance based on the headers and $_SERVER data.
+     *
+     * @param array<string, string|list<string>> $headers
+     * @param array $server SAPI parameters
```

Comment:
Psalm has a quite decent coverage of `$_SERVER`: https://github.com/vimeo/psalm/blob/7869cb548a5d88399f971ce9f897fe83abbd4213/src/Psalm/Internal/Analyzer/Statements/Expression/Fetch/VariableFetchAnalyzer.php#L671

Seems like it's an `array<non-empty-string, list<string>|int|float|string>>` from there, but I'm a bit groggy, so it might be worth double-checking